src/lxc/lxc_init.c

```diff
@@ -92,6 +92,86 @@ static struct arguments my_args = {
 	.shortopts = short_options
 };
 
+static void prevent_forking(void)
+{
+	FILE *f;
+	char name[PATH_MAX], path[PATH_MAX];
+	int ret;
+
+	f = fopen("/proc/self/cgroup", "r");
+	if (!f) {
+		SYSERROR("opening /proc/self/cgroup");
+		return;
+	}
+
+	while (!feof(f)) {
+		int fd;
+
+		if (2 != fscanf(f, "%*d:%s:%s", name, path)) {
```

Comment:
Any concerns about length of path?

Reply:
Anwyay, the only change was to this line, and it's now updated.

Reply:
... do you trust PATH_MAX?

Reply:
Sure, it's POSIX. If you can't trust the headers, none of this works.